$namespaces:
  edam: http://edamontology.org/
$schemas:
- https://edamontology.org/EDAM_1.18.owl
arguments:
- position: -1
  valueFrom: ApplyBQSR
baseCommand:
- /gatk/gatk
class: CommandLineTool
cwlVersion: v1.0
doc: "This tool implements the Genome Analysis Toolkit (GATK) function \"ApplyBQSR,\"\
  \ \nwhich is used for base quality score recalibration in genomic data processing.\
  \ \nIt takes as input a BAM file, a base recalibration file, and various optional\
  \ \nparameters, and produces a recalibrated BAM file along with optional index and\
  \ \nVCF outputs.\n"
hints:
  DockerRequirement:
    dockerPull: quay.io/biocontainers/gatk4:4.1.1.0--0
  SoftwareRequirement:
    packages:
      gatk:
        specs:
        - https://identifiers.org/biotools/gatk
        - https://anaconda.org/bioconda/gatk4
        version:
        - 4.1.1.0
inputs:
  BaseRecalFile:
    doc: Base recalibration file generated by BaseRecalibrator.
    inputBinding:
      prefix: --bqsr-recal-file
    type: File
  InputFile:
    doc: Input BAM file to be recalibrated.
    inputBinding:
      prefix: --input
    type: File
  JavaOptions:
    doc: Java options for memory and other settings.
    inputBinding:
      position: -2
      prefix: --java_options
    type: string?
  Output:
    default: ApplyBQSR.bam
    doc: Name of the output recalibrated BAM file.
    inputBinding:
      prefix: --output
      valueFrom: ApplyBQSR.bam
    type: string
  ReadFilter:
    doc: Read filters to apply during processing.
    inputBinding:
      itemSeparator: ','
      prefix: --read-filter
    type: string[]?
  isCreateBamIndex:
    doc: Create a BAM index for the output file.
    inputBinding:
      prefix: --create-output-bam-index
    type: boolean?
  isEmitOriginalQuals:
    doc: Emit the original base quality scores.
    inputBinding:
      prefix: --emit-original-quals
    type: boolean?
label: GATK ApplyBQSR
outputs:
  alignment:
    doc: Recalibrated BAM file.
    outputBinding:
      glob: ApplyBQSR.bam
    type: File
  index:
    doc: Index file for the recalibrated BAM, if created.
    outputBinding:
      glob: ApplyBQSR.bai
    type:
    - 'null'
    - File
  vcf:
    doc: VCF file output, if applicable.
    format: edam:format_3016
    outputBinding:
      glob: '*.vcf'
    type:
    - 'null'
    - File
requirements:
  InlineJavascriptRequirement: {}
  ResourceRequirement:
    ramMin: 1024
